src/nnet3/nnet-compile-utils.cc

```diff
@@ -598,5 +598,44 @@ bool HasContiguousProperty(
   return true;
 }
 
+
+// see comment in header.
+void GetNxList(const std::vector<Index> &indexes,
+               std::vector<std::pair<int32, int32> > *pairs) {
+  // set of (n,x) pairs
+  std::unordered_set<std::pair<int32, int32>, PairHasher<int32> > n_x_set;
+
+  for (std::vector<Index>::const_iterator iter = indexes.begin();
+       iter != indexes.end(); ++iter)
+    n_x_set.insert(std::pair<int32, int32>(iter->n, iter->x));
+  pairs->clear();
+  pairs->reserve(n_x_set.size());
+  for (std::unordered_set<std::pair<int32, int32>, PairHasher<int32> >::iterator
+           iter = n_x_set.begin(); iter != n_x_set.end(); ++iter)
+    pairs->push_back(*iter);
+  std::sort(pairs->begin(), pairs->end());
+}
+
+
+// see comment in header.
+void GetTList(const std::vector<Index> &indexes,
+              std::vector<int32> *t_values) {
+  // set of t values
+  std::unordered_set<int32> t_set;
```

Comment:
In the normal case, there could be many (e.g. 128) copies of each 't' value, so in that case I think
the way we have it is more efficient.  (Also Index is a struct, not an integer).